validator.go

```diff
@@ -179,3 +189,20 @@ func WithAllowUnknownFields() ValidatorOption {
 		c.allowUnknownFields = true
 	}
 }
+
+type validationConfig struct {
+	failFast bool
```

Comment:
Do we want to permit setting failFast at Validate call-time? I think that would be a nice flexibility now that it's included in the config.

Reply:
In order to do that we do need a bit of refactoring, since there's no obvious way to implement that as-is. I've chosen to follow what connect-go does, turning both types of options into interfaces, having a union interface for options that can be applied in both circumstances, and then have all of the options implemented as structs. Since this is really cluttering the `validator.go` file I also moved it to `option.go`, also like connect-go.